File: frontend/src/types/detection.ts
export interface Detection {
  id: number;
  detection_uid: string;
  camera_id: number;
  camera_code: string;
  camera_name: string;
  vehicle_id?: number | null;
  vehicle_uid?: string | null;
  plate_id?: number | null;
  plate_number?: string | null;
  timestamp: string;
  ocr_confidence?: number | null;
  vehicle_color?: string | null;
  vehicle_type?: string | null;
  direction_travel?: string | null;
  snapshot_path?: string | null;
  plate_anomaly_flags?: Record<string, unknown> | null;
  processing_metadata?: Record<string, unknown> | null;
  association_confidence?: number | null;
  is_simulated: boolean;
  created_at: string;
}

export interface DetectionListResponse {
  total: number;
  items: Detection[];
}

File: backend/app/schemas/detection.py
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, ConfigDict, Field


class DetectionResponse(BaseModel):
    id: int
    detection_uid: str
    camera_id: int
    camera_code: str
    camera_name: str
    vehicle_id: Optional[int] = None
    vehicle_uid: Optional[str] = None
    plate_id: Optional[int] = None
    plate_number: Optional[str] = None
    timestamp: datetime
    ocr_confidence: Optional[float] = None
    vehicle_color: Optional[str] = None
    vehicle_type: Optional[str] = None
    direction_travel: Optional[str] = None
    snapshot_path: Optional[str] = None
    plate_anomaly_flags: Optional[Dict[str, Any]] = None
    processing_metadata: Optional[Dict[str, Any]] = None
    association_confidence: Optional[float] = None
    is_simulated: bool = True
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class DetectionListResponse(BaseModel):
    total: int
    items: List[DetectionResponse]
